Lib/test/test_string.py

```diff
@@ -270,6 +270,10 @@ def test_invalid_placeholders(self):
         raises(ValueError, s.substitute, dict(who='tim'))
         s = Template('$who likes $100')
         raises(ValueError, s.substitute, dict(who='tim'))
+        # Template.idpattern should match to only ASCII characters.
+        # https://bugs.python.org/issue31672
+        s = Template("$who likes $ı")  # (0x131, DOTLESS I)
```

Comment:
Test also `'İ'` (0x130). `'İ'.lower() == 'i'`. In older Python versions `[a-z]` didn't match `'ı'`, but matched `'İ'`.